### PR TITLE
feat(repo): Restart this project, create new configurations

### PR DESCRIPTION
**Pull request by bryvamper**
This is an effort to modernize and revisit our code style for Typescript projects. It also includes a bundled Prettier configuration. Take a lot at the README for simple setup instructions.

I'm going to publish a new version on NPM so we can play with the config in our repos. For now, using mostly recommended configs for typescript (with a couple overrides) and using prettier for all the stylistic settings.

The default export for this package is the Node eslint config. This repo can hold multiple configs, but we'll be testing the react config first.

**Why not AirBnB or some other opinionated point?**
Looking at the rules in these kinds of plugins, stylistically they are all mostly supported by Prettier with sensible defaults. I'm using eslint typescript recommended defaults and they look pretty good. I think we can add rules as we go that **really matter in preventing bugs.** 

**Note:**
This repo is locked down to eslint v5 since the typescript eslint plugin is not yet compatible with v6. At a later time we will update the dependencies.

### CODE REVIEW

**Review by stton — APPROVED** [review]
👍 looks like a good starting point for our new configuration. Will be best to start using this config in our repos to get a feel and have more context behind adding new rules / tweaking the configuration. While we can do this by npm linking the project and updating some local configs in repo, it will be easiest to publish an early version to test out since this configuration is out of date and not being used anywhere at the moment 💀

File: base.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    parserOptions: {
+        ecmaVersion: 6, // Allows for the parsing of ES6 features
+        sourceType: 'module', // Allows for the use of imports
+    },
+    rules: {
+        // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+        // Public is assumed by default for class members, and shouldn't be explicit
+        '@typescript-eslint/explicit-member-accessibility': 'no-public',
+        // It's often not feasible to avoid an "any" type for some siuations
+        '@typescript-eslint/no-explicit-any': 'off',
```
Comment (davidungio) [inline base.js]: Thoughts on putting this to `warn`?

**Review by davidungio — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -1,18 +1,20 @@
 {
   "name": "eslint-config-videoamp",
   "version": "4.2.0",
```
Comment (davidungio) [inline package.json:3]: We'll need to bump this up to a major version

**Comment by josh803316** [comment]
Thanks for updating this, was also causing an issue in release-info :)

File: package.json (hunk 2 above)
Comment (bryvamper) [reply package.json:3]: Yes, was planning on that

File: base.js (hunk 1 above)
Comment (bryvamper) [reply base.js]: We can try that out, sure.

**Comment by davidungio** [comment]
@bvamp @josh803316 - I've removed SemaphoreCI from this project. Please ignore the build fails

**Comment by josh803316** [comment]
@davidungio ack

File: base.js (hunk 1 above)
Comment (bryvamper) [reply base.js]: Changed to warn

**Pull request merged**